pipelines/api/ghauth.py

```diff
@@ -184,16 +184,26 @@ def cb(*args):
 
 
             allowed_teams = self.settings['auth'].get('teams', [])
-            log.debug('Allowed teams: %s' % allowed_teams)
+            allowed_teams_set = set()
+            for org, team in allowed_teams:
+                allowed_teams_set.add((org.lower(), team.lower()))
+            log.debug('Allowed teams: %s' % allowed_teams_set)
             teams = r.json()
 
-            for org, team in allowed_teams:
-                if filter(lambda x: x['slug'].lower() == team.lower() and x.get('organization',{}).get('login', '').lower() == org.lower(), teams):
-                    log.debug('Allowed access to github user for team %s/%s' % (org, team))
+            user_teams_set = set()
+            for x in teams:
+                o = x.get('organization',{}).get('login', '').lower()
+                t = x.get('slug', '').lower()
+                user_teams_set.add((o, t))
```

Comment:
if `o` or `t` == `''` in here. then no need to add them to user_teams_set I guess